std/src/main/scala/algebra/std/array.scala

```diff
@@ -0,0 +1,42 @@
+package algebra
+package std
+
+import scala.{specialized ⇒ sp}
+
+package object array extends ArrayInstances
+
+trait ArrayInstances {
+  implicit def arrayEq[@sp A: Eq]: Eq[Array[A]] = new ArrayEq[A]
```

Comment:
these should probably have the same `@sp(Int, Float, Long, Double)` that we standardized on.